modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/claims-settled/index.js

```diff
@@ -0,0 +1,463 @@
+						settledClaimsThreeMonths?.items.length
+					);
+
+					handleBuildLegend(
+						getProductNameOfClaims(settledClaimsThreeMonths)
+					);
+				});
+
+				getClaimsByPeriodSettled(
+					currentDateString[0],
+					currentDateString[1],
+					currentDateString[2],
+					threeMonthsAgoDate[0],
+					threeMonthsAgoDate[1],
+					threeMonthsAgoDate[2]
+				).then((results) => {
+					const activeClaimsThreeMonths = results?.data?.totalCount;
+
+					setTotalOfClaimsThreeMonths(activeClaimsThreeMonths);
+				});
+			}
+
+			if (selectedFilterDate === PERIOD.SIX_MONTHS) {
+				getSettledClaims(
+					currentDateString[0],
+					currentDateString[1],
+					currentDateString[2],
+					sixMonthsAgoDate[0],
+					sixMonthsAgoDate[1],
+					sixMonthsAgoDate[2]
+				).then((results) => {
+					const settledClaimsSixMonths = results?.data;
+
+					setSixMonthsSettledClaims(
+						settledClaimsSixMonths?.items.length
+					);
+
+					handleBuildLegend(
+						getProductNameOfClaims(settledClaimsSixMonths)
+					);
+				});
+
+				getClaimsByPeriodSettled(
+					currentDateString[0],
+					currentDateString[1],
+					currentDateString[2],
+					sixMonthsAgoDate[0],
+					sixMonthsAgoDate[1],
+					sixMonthsAgoDate[2]
+				).then((results) => {
+					const activeClaimsSixMonths = results?.data?.totalCount;
+
+					setTotalOfClaimsSixMonths(activeClaimsSixMonths);
+				});
+			}
+
+			if (selectedFilterDate === PERIOD.YTD) {
+				getSettledClaims(
+					currentDateString[0],
+					currentDateString[1],
+					currentDateString[2],
+					currentDateString[0],
+					january,
+					january
+				).then((results) => {
+					const settledClaimsYearToDate = results?.data;
+
+					setYearToDateSettledClaims(
+						settledClaimsYearToDate?.items.length
+					);
+
+					handleBuildLegend(
+						getProductNameOfClaims(settledClaimsYearToDate)
+					);
+				});
+
+				getClaimsByPeriodSettled(
+					currentDateString[0],
+					currentDateString[1],
+					currentDateString[2],
+					currentDateString[0],
+					january,
+					january
+				).then((results) => {
+					const activeClaimsYearToDate = results?.data?.totalCount;
+
+					setTotalOfClaimsYearToDate(activeClaimsYearToDate);
+				});
+			}
+
+			setLoadData(true);
+		});
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [selectedFilterDate]);
+
+	const chartLoadData = [
+		{
+			dataColumns: [
+				['settled', thisMonthSettledClaims],
+				['remaining', totalOfClaimsThisMonth - thisMonthSettledClaims],
+			],
+			percentual: getSettledPercentual(
+				thisMonthSettledClaims,
+				totalOfClaimsThisMonth
+			),
+			period: 1,
+		},
+		{
+			dataColumns: [
+				['settled', threeMonthsSettledClaims],
+				[
+					'remaining',
+					totalOfClaimsThreeMonths - threeMonthsSettledClaims,
+				],
+			],
+			percentual: getSettledPercentual(
+				threeMonthsSettledClaims,
+				totalOfClaimsThreeMonths
+			),
+			period: 2,
+		},
+		{
+			dataColumns: [
+				['settled', sixMonthsSettledClaims],
```

Comment:
You are using this word `settled `more than once. Could you put this as a constant?

Reply:
Done!